crates/router/src/core/payouts/helpers.rs

```diff
@@ -433,121 +440,246 @@ pub async fn get_or_create_customer_details(
+        api::ConnectorChoice::Decide,
+        api::ConnectorChoice::StraightThrough,
     ))
 }
 
 pub fn should_call_payout_connector_create_customer<'a>(
     state: &AppState,
-    connector: &api::PayoutConnectorData,
+    connector: &api::ConnectorData,
     customer: &'a Option<domain::Customer>,
     connector_label: &str,
 ) -> (bool, Option<&'a str>) {
     // Check if create customer is required for the connector
-    let connector_needs_customer = state
-        .conf
-        .connector_customer
-        .payout_connector_list
-        .contains(&connector.connector_name);
-
-    if connector_needs_customer {
-        let connector_customer_details = customer.as_ref().and_then(|customer| {
-            get_connector_customer_details_if_present(customer, connector_label)
-        });
-        let should_call_connector = connector_customer_details.is_none();
-        (should_call_connector, connector_customer_details)
-    } else {
-        (false, None)
+    match PayoutConnectors::try_from(connector.connector_name) {
+        Ok(connector) => {
+            let connector_needs_customer = state
+                .conf
+                .connector_customer
+                .payout_connector_list
+                .contains(&connector);
+
+            if connector_needs_customer {
+                let connector_customer_details = customer.as_ref().and_then(|customer| {
+                    get_connector_customer_details_if_present(customer, connector_label)
+                });
+                let should_call_connector = connector_customer_details.is_none();
+                (should_call_connector, connector_customer_details)
+            } else {
+                (false, None)
+            }
+        }
+        _ => (false, None),
     }
 }
 
+pub async fn get_gsm_record(
+    state: &AppState,
+    error_code: Option<String>,
+    error_message: Option<String>,
+    connector_name: Option<String>,
+    flow: String,
+) -> Option<storage::gsm::GatewayStatusMap> {
+    let get_gsm = || async {
+        state.store.find_gsm_rule(
+                connector_name.clone().unwrap_or_default(),
+                flow.clone(),
+                "sub_flow".to_string(),
+                error_code.clone().unwrap_or_default(), // TODO: make changes in connector to get a mandatory code in case of success or error response
+                error_message.clone().unwrap_or_default(),
+            )
+            .await
+            .map_err(|err| {
+                if err.current_context().is_db_not_found() {
+                    logger::warn!(
+                        "GSM miss for connector - {}, flow - {}, error_code - {:?}, error_message - {:?}",
+                        connector_name.unwrap_or_default(),
+                        flow,
+                        error_code,
+                        error_message
+                    );
+                    metrics::AUTO_RETRY_GSM_MISS_COUNT.add(&metrics::CONTEXT, 1, &[]);
```

Comment:
We should have separate metrics for payouts, it can't be same as Payment metrics